### ISSUE TITLE
CiceroMark<->OOXML transformers

### ISSUE THREAD

**Issue by K-Kumar-01** [accordproject/markdown-transform#397]
# Feature Request 🛍️

Incorporating the CiceroMark->OOXML transformer and improving the currently implemented OOXML->CiceroMark transformer.

## Use Case

It will allow the conversion of Docx files into CiceroMark JSON and vice versa. In addition, using this `Docx` files can be converted into different other formats like `PDF` or `HTML` as well.

## Possible Solution

The transformer for OOXML->CiceroMark is already implemented. It needs to be updated with some entities to allow full transformation.

The transformer for CiceroMark->OOXML is created in `cicero-word-add-in` branch. It needs to be transferred/transported here with some changes to allow the transformation.

## Detailed Description

Currently, the CiceroMark->OOXML transformer supports the following conversions:
| CiceroMark Entity  | OOXML Entity |
| --------  | -------- |
| Text      | `<w:t>` |
| Paragraph     | `<w:p>Content</w:p>` |
| Linebreak      | `<w:p/>` |
| Softbreak      | `<w:r><w:sym></w:r>` |1
| Emph     | `<w:i>` |
| Variable      | `<w:sdt>` |
| List Block/ List     | `<w:numPr><w:num w:val={ordered/unordered}/></w:numPr>` |
| List Item(Text)     | `<w:t/>` |
| List Item(Variable)     | `<w:sdt/>` |

Conversions which are left:

- `Formula / Ergo expressions`
- `Clause`
- `Conditional`
- `Optional`
- `Strong`
- `BlockQuote`
- `CodeBlock`
- `Code`
- `ThematicBreak`
- `Link`
- `Image`

In the left conversions, we need to decide which ones need major importance/priority and which can be given a lower priority. Furthermore, we also need to think about whether all these will be present in the contract (IMO, Code and CodeBlock generally won't occur in the contract)

**Comment by algomaster99** [comment]
> Formula / Ergo expressions
Clause
Conditional
Optional
Strong
BlockQuote
CodeBlock
Code
ThematicBreak
Link
Image

Create a checkbox for this. It will be easier to track.

> we also need to think about whether all these will be present in the contract (IMO, Code and CodeBlock generally won't occur in the contract)

Do raise this up in the meeting tomorrow :)

Since you have created the table, don't forget to link this to the next wiki page you will write.

**Comment by algomaster99** [comment]
> Entities and their corresponding Ciceromark

Will you also be maintaining the OOXML counterpart? I recommend so.

**Comment by K-Kumar-01** [comment]
> > Entities and their corresponding Ciceromark
> 
> Will you also be maintaining the OOXML counterpart? I recommend so.

Yeah, I will also maintain them as we start writing the transformer for it.

**Comment by K-Kumar-01** [comment]
@algomaster99 @dselman 
A doubt regarding the variable transformer.
![Screenshot from 2021-06-16 17-12-25](https://user-images.githubusercontent.com/59891164/122213047-27b5bb80-cec6-11eb-8a0f-803ebb160f11.png)
When we rendered in the add-in, some variables had extra quotation marks. For more reference see [here](https://github.com/accordproject/cicero-word-add-in/issues/39). The quotation marks were removed.

The tests now require these quotation marks in the transformer, as shown in the above image(test failing if no quotation marks). 

So how to proceed here? 
I have come on two solutions:
1. Remove the stripping off quotation marks altogether (rendering the probably unnecessary quotes).
2. Remove the stripping off quotation marks in the transformer to make it pass the test. The problem here can be if we convert from CiceroMark->OOXML via the transformer or markus-cli (possibly in the future) and then open the file in the add-in, we will have double quotes in variable values. In addition, if we add another template using the add-in, those values won't have double quotes.

PS: I am not a fan of either approach.

**Comment by algomaster99** [comment]
Where did you get the CiceroMark for testing? If you got it from a latest template, change the variable transformer to enclose variables in `""`.

**Comment by K-Kumar-01** [comment]
@algomaster99 
I got it from `acceptance-of-delivery.json`.
As for variable transformer do you mean `CiceroMarkToOOXML` or `OOXMLtoCiceroMark`.
We strip off quotation marks in the former one.
Also, enclosing variable values in `""` in the latter won't make sense as variables which are of type `DateTime, Number, etc` don't have quotes around them.

**Comment by algomaster99** [comment]
First of all, that `acceptance-of-delivery.json` is parsed from an older version of the template. You might want to update that. Next, if the latest parsed CiceroMark still encloses its variables in `""`, you don't need to add extra code for enclosing the value in `""` or stripping them off `""` because when you will be iterating through the CiceroMark, you will get the value as `"\"Party A\" "` and not `"Party A"`. But if **does not** enclose the value in `""`, again, you don't need to add extra code because you will get `"Party A"` as the value only and not `"\"Party A\""`.

**Comment by K-Kumar-01** [comment]
@algomaster99 
>Overall, I don't see why we should be concerned about explicitly stripping or adding "" around the variable value.

The main reason was that some variable values had `""`wrapped and some not. The `""`was wrapped around basically in strings. So I thought it might have been unintentional and that they could have been inserted while transforming by mistake.

**Comment by K-Kumar-01** [comment]
@algomaster99 
I will keep the values as it is in the transformers

**Comment by algomaster99** [comment]
Exactly my point :)

**Comment by algomaster99** [comment]
@K-Kumar-01 from now, don't mention issue numbers in every commit you push. As you can see, it has cluttered this PR. I think mentioning the issue number should only be necessary for the PR title as your PRs are squashed anyway.

**Comment by K-Kumar-01** [comment]
@algomaster99 @dselman 
There is no such thing as `blockquote` in ms-word. It basically involves styling a given paragraph. [Reference videos](https://www.youtube.com/results?search_query=blockquote+in+ms-word), [Styling Reference for shading](https://www.officetooltips.com/word_365/tips/how_to_use_paragraph_borders_and_shading.html).

So from these, can you decide some specifications for the blockquote which we need?

Also, for `inline code, I am thinking of this formatting:
![Screenshot (70)](https://user-images.githubusercontent.com/59891164/123130580-7e913700-d46a-11eb-8a7c-f3f9bdf48f7a.png)

Thanks in advance:)
